### ISSUE TITLE
Can not deploy private registry docker images

### ISSUE THREAD

**Issue by r-chris** [nulldriver/cf-cli-resource#24]
Manifest file:
```
applications:
- name: test-deploy-private
  docker:
    image: PRIVATE_DOCKER_REGISTRY_NAME/test-app:latest
    username: ***
  command: null
  memory: 128M
  instances: 1
  disk_quota: 5G
  health-check-type: process
  routes:
    - route: test-deploy-private.***
```

Deployment plan:
```
- name: deploy-cli-private
  plan:
  - get: git
    passed: [build-private]
    trigger: false
  - put: cf-cli-env
    params:
      command: push
      manifest: git/manifest-private.yml
      environment_variables:
        CF_DOCKER_PASSWORD: ((docker-registry-password))
      cf_trace: true
```

Error message:
```
Using manifest file git/manifest-private.yml

FAILED
No Docker password was provided. Please provide the password by setting the CF_DOCKER_PASSWORD environment variable.
```

**Comment by r-chris** [comment]
Probably related to #23. I tried to create a very simple, minimalistic deployment setup to test pushing an app from our private registry, but it doesn't seem to work with the build-in cf-resource or with this cf-cli-resource. Any help would be much appreciated.

**Comment by patrickcrocker** [comment]
Implemented in [v2.8.0](https://github.com/nulldriver/cf-cli-resource/releases/tag/v2.8.0).

Please note that you are going to see the same issue you [reported here](https://github.com/concourse/cf-resource/issues/48) if you use a placeholder for `docker.username` in your `manifest.yml` file.

This is a limitation of the `cf` cli, and not the `cf-cli-resource` or `cf-resource`.
